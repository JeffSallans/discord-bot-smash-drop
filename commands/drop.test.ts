import * as seed from 'seed-random';
seed('123', { global: true });

let importedCommands, importedReadyForCommands, destroy;

describe('drop', () => {
  let mockMessageObject;

  beforeAll(() => {
    const { commands, readyForCommands, destroy } = require('..');
    importedCommands = commands;
    importedReadyForCommands = readyForCommands;
    mockMessageObject = {};
    mockMessageObject.reply = (message) => {
      expect(message).toMatchSnapshot();
    }
  });

  afterAll(async () => {
    await destroy();
  });

  it('works with no arguments', async () => {
    await importedReadyForCommands;
    await importedCommands['drop'](mockMessageObject, ['drop']);
  }, 10000);

  it('works with player args', async () => {
    await importedReadyForCommands;
    await importedCommands['drop'](mockMessageObject, ['drop', 'test']);
  }, 10000);
});
