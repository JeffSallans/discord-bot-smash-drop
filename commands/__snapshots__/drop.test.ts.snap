// Jest Snapshot v1, https://goo.gl/fbAQLP
exports[`drop works with no arguments 1`] = `
"
:Rare::Gnaw:  :Rare::Ayla:  :Epic::Derpl:  :Rare::Swiggins:  :Epic::Skree: -- Drop #1 cgs
:Rare::Jimmy:  :Rare::Qitara:  :Rare::Yoolip:  :Rare::VinnieandSpike:  :Epic::Ksenia: -- Drop #3 hasp
:Epic::Coco:  :Rare::Froggy:  :Rare::Gnaw:  :Rare::Penny:  :Rare::Ted: -- Drop #2 mathmatical
:Epic::Derpl:  :GoldenRare::Lonestar:  :Rare::MaxFocus:  :Rare::Ksenia:  :Rare::Ted: -- Drop #4 mlripper
:Rare::Derpl:  :Rare::Penny:  :Rare::SnorkGunk:  :Epic::Gnaw:  :Rare::Clunk: -- Drop #6 qazwode
:Rare::Lonestar:  :Rare::Swiggins:  :GoldenEpic::MaxFocus:  :Rare::SnorkGunk:  :Rare::Penny: -- Drop #5 thynix
:Rare::Scoop:  :Rare::Penny:  :Rare::Froggy:  :Epic::Skolldir:  :Rare::SnorkGunk: -- Drop #7 test "
`;

exports[`drop works with player args 1`] = `
"
:Rare::Scoop:  :Rare::Penny:  :Rare::Froggy:  :Epic::Skolldir:  :Rare::SnorkGunk: -- Drop #7 test "
`;
